sections/multicolumn.liquid

```diff
@@ -81,17 +81,29 @@
                       style="padding-bottom: {{ 1 | divided_by: highest_ratio | times: 100 }}%;"
                     {% endif %}
                   >
-                    {%- capture sizes -%}
-                      (min-width: 990px) {% if section.blocks.size <= 2 %}710px{% else %}550px{% endif %}, (min-width:
-                      750px) {% if section.blocks.size == 1 %}710px{% else %}550px{% endif %}, calc(100vw - 30px)
-                    {%- endcapture -%}
+                    {%- liquid
+                      assign number_of_columns = section.settings.columns_desktop
+                      assign number_of_columns_mobile = section.settings.columns_mobile
+                      assign max_image_width = settings.page_width | times: 2
+                      if section.settings.image_width == 'half'
+                        assign image_width = 0.5
+                      elsif section.settings.image_width == 'third'
+                        assign image_width = 0.33
+                      else
+                        assign image_width = 1
+                      endif
+                    -%}
+                    {% capture sizes %}
+                      (min-width: 990px) {{ settings.page_width | times: image_width | divided_by: number_of_columns | round: 0 }}px,
+                      calc(100vw * {{ image_width }} / {{ number_of_columns_mobile }})
+                    {% endcapture %}
                     {{
                       block.settings.image
-                      | image_url: width: 1420
+                      | image_url: width: max_image_width
                       | image_tag:
                         loading: 'lazy',
+                        widths: '50, 75, 100, 150, 200, 300, 400, 500, 750, 1000, 1500, 2000, 2500, 3000',
```

Comment:
Interested in what other devs think about going this route as it should be something we do moving forward. Assuming no one objects to it, I wonder if having a smaller increment than 500px for the larger images is worthwhile since those are the ones with the most cost. Maybe 250px or something?

Reply:
Thanks for the feedback. I added a few more numbers. Also increased the largest one to 3250 to cover an edge case when page-width is 1600px and column number for desktop is 1.